src/main/java/org/opensearch/flowframework/util/WorkflowTimeoutUtility.java

```diff
@@ -205,8 +208,10 @@ public static void fetchWorkflowStateAfterTimeout(
         client.execute(
             GetWorkflowStateAction.INSTANCE,
             new GetWorkflowStateRequest(workflowId, false, tenantId),
-            ActionListener.wrap(
-                response -> listener.onResponse(new WorkflowResponse(workflowId, response.getWorkflowState())),
+            ActionListener.wrap(response -> {
+                TenantAwareHelper.releaseProvision(tenantId);
```

Comment:
Are we sure we want to release provisioning, incase provisioning is still ongoing?

Reply:
Good callout. No. I just reverted the last commit, will post a comment below outlining entire code path.